### PR TITLE
master

### PR DESCRIPTION
**Pull request by tjheffner**
here's the first version of the pattern lab workshopper

future:
-- display the 'example.js' file for each problem to correctly show our solutions, differs from the console output tests in solution.js. currently doesn't do that well, but the files are included anyways
-- update some of the lesson text. 
-- more lessons/separate workshopper for more advanced PL concepts?

### CODE REVIEW

**Comment by robbiethegeek** [comment]
WOW... will try to start working through this AMAZING pull request and provide feedback!

File: menu.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+    "CREATE AN ATOM"
+  , "MORE ATOMS"
+  , "MAKE IT MOLECULAR"
+  , "ORGANISMS"
+  , "TEMPLATES"
+  , "PAGES"
+  , "JSON"
+  , "SIDECAR JSON"
+  , "LET'S GET SASSY"
+]
```
Comment (robbiethegeek) [inline menu.json]: @tjheffner can you go through and fix all the new line at end of file style warnings.

File: problems/create_an_atom/verify.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env node
+
+var exec = require('child_process').exec;
+var fs = require('fs');
+var path = require('path');
+var filename = "01-tutorial-button.mustache";
+
+//check the directory exists
```
Comment (robbiethegeek) [inline problems/create_an_atom/verify.js:8]: Let's remove the commented code.

File: problems/make_it_molecular/verify.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+function findFile() {
+  if (process.cwd().match("molecules")) {
+    check(process.cwd())
+  } else {
+    check(path.join(process.cwd(), "/molecules/"))
+  }
+
+  function check(userspath) {
+    fs.readdir(userspath, function(err, files) {
+      if (err) return console.log(err);
```
Comment (robbiethegeek) [inline problems/make_it_molecular/verify.js]: in general I would be more verbose and not do a one-liner if statement.  I would say that about all single statements that are if's or else's throughout the file.

File: problems/create_an_atom/verify.js (hunk 2 above)
Comment (tjheffner) [inline problems/create_an_atom/verify.js]: took out the legacy commented code from the git-it example

File: menu.json (hunk 1 above)
Comment (tjheffner) [inline menu.json]: done

File: problems/make_it_molecular/verify.js (hunk 3 above)
Comment (tjheffner) [inline problems/make_it_molecular/verify.js]: I'm not quite sure how to best update these to be more verbose...

A lot of them exist/are structured that way so the workshopper can check things incrementally/match the console logs, and the node.js docs recommend including "if (err) return console.log(err);" when working with fs and files or data.

Comment (robbiethegeek) [inline problems/make_it_molecular/verify.js]: What I am alluding to is 

if (err) return console.log(err);

Is valid js there is a lot of style guides that talk about the right or wrong style:
https://github.com/RisingStack/node-style-guide#blocks

**Comment by robbiethegeek** [comment]
@illepic / @EvanLovely  When you get a chance can you guys take a pass at the Pattern Lab perspective.  @tjheffner did @illepic talk to you about updating the workshopper-jlord and abstract away the git-it stuff?

**Comment by tjheffner** [comment]
He did, and we did. It happened in this commit: https://github.com/tjheffner/pattern-lab-workshop/commit/44fe9c8347c136ddc1e0c81badc01c2d97459f60

and this is the "new" workshopper with updated output changes: https://github.com/tjheffner/workshopper/tree/master

Currently workshopper-jlord points to my updated one, if/until jlord merges the updates

Now anyone using the "new" workshopper framework can spit out any strings they want when a user selects/passes a lesson, not just things related to git-it

**Comment by robbiethegeek** [comment]
@EvanLovely @illepic what is the status of this PR?  would love to get @tjheffner any other feedback and move this from a glimmer in my / bloom's eye to a real released workshop that I can add to nodeschool!

**Comment by robbiethegeek** [comment]
BUMP!! :)

**Comment by EvanLovely** [comment]
I don't have enough bandwidth to review stuff on this project as my attention's on PL Starter & all the P2 Yeoman generators for Dev Tools; sorry! I'd be happy to look over @illepic's shoulder as he reviews and give my 2¢ though!

Honestly, given the young nature of this codebase, I don't see why we couldn't merge @tjheffner's stuff in as it's obvious he's given it a lot of thought and work; nice job Tanner!

**Comment by robbiethegeek** [comment]
I am going to merge this in @illepic if you want this updated please create an issue and let us know I am going to create a pull request to add this to nodeschool's page.

**Comment by EvanLovely** [comment]
@robbiethegeek ++

**Pull request merged**